app/services/listing_index_service/search/sphinx_adapter.rb

```diff
@@ -105,10 +113,15 @@ def search_with_sphinx(community_id:, search:, included_models:)
           star: true,
           with: with,
           with_all: with_all,
-          order: 'sort_date DESC'
+          order: 'sort_date DESC',
+          max_query_time: 500 # Timeout and fail after 500ms
         )
 
-        {count: models.total_entries, listings: models}
+        begin
+          success_result(models.total_entries, models, includes)
+        rescue ThinkingSphinx::SphinxError => e
+          Result::Error.new(e)
```

Comment:
I haven't tested the indexing thing but this is what the Sphinx-documentation says, all errors raised are SphinxErrors. The case of sphinx not running in development is just a connection error, so it's indistinguishable from a network error in production or production sphinx being down.